### DISCUSSION TITLE
`Terminal` does not provide a way to monitor process completion?

### DISCUSSION THREAD

**Discussion by tinder-tannerbennett** [microsoft/vscode-discussions#861] [Extension Development QnA] [answered]
I want to have a command in my extension that runs a process with either a progress indicator or by showing the terminal output itself (the user can choose).

I need to run some code when the process finishes. Using `exec`, I am able to do this, but using `Terminal`, there is no obvious way to do something when an input command finishes.

Is there a way to say, start a command with `exec` and pass it to a `Terminal` or something? Or some way I can monitor the running task from the terminal's `processId`?

**Comment by bwateratmsft** [comment, marked as answer]
What we do in the Docker extension might be helpful here: https://github.com/microsoft/vscode-docker/blob/74bc9a562ed5f8a4ce87622a0bf63e8df1b7f933/src/runtimes/runners/TaskCommandRunnerFactory.ts#L38-L83

Basically we execute things as a VSCode task which lets the user see the terminal output but gives us the exit code

**Reply by tinder-tannerbennett** [reply]
That is _extremely_ helpful, thank you!
